### ISSUE TITLE
[A11y] Universal Viewer

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#1992]
### Operating System

Windows

### Browser

Edge

### Affected page

Work Show

### Admin user?

Yes

### Issue Description

Example page: compound parent: https://staging.oregondigital.org/concern/generics/df727084j

As the universal viewer isn't our tool, I'm going to try and collect all failures unique to it here. That way we can figure out what, if anything, can be done to improve it.

- I cannot get to the index or thumbnail links via keyboard
- There are places I lose my focus indicator, but it doesn't seem to be consistent; I think JS adds and removed phantom elements at times.
- One area that *seems* like it loses focus is actually a useful one: tabbing past the settings / gear icon, there's no visual focus indicator, but the focus is in the main viewing area. While here you can use the keyboard to pan and zoom, which is critically important **if** you know that's where your focus is.
- We should provide a link of keyboard shortcuts - for instance, the arrow keys pan, plus and minus zoom.
- Disabled buttons don't announce that they're disabled
- Activating navigation buttons (previous page, next page, etc.) changes what's in the viewport, but doesn't announce anything to a screen reader user. The image viewer functionality could be considered to require sight, so it may not be a hard failure.
- It might be helpful to have a link to skip past the viewer. Users with no sight or very poor sight may care about the metadata, but not about the image viewer.
  - Along these lines, a heading here could help, too. If you know by the heading structure where the image viewer lies relative to other data, you can more easily skip to or past it using a screen reader or even a browser plugin.
- The "more information" widget is actually two links, but only one can be tabbed to, which is confusing if using something that lists links
- Many controls are links and should be buttons since the "links" don't actually go anywhere
- The "search within this item" button (magnifying glass icon) is a link instead of a button for some reason, and has no accessible name
- Errors are not identified well - when typing a bad page number, for instance, the dialog that shows up is almost completely inaccessible
- Disabled controls can receive focus but have poor contrast

### Suggested Fix

I don't know if this can be fixed on our end or not.

### Can anybody QA or does this need an expert?

Needs expert

### QA Steps

Depends on fix, if one is even feasible.

**Comment by azle** [comment]
Updated 12/16/2024 IN PROGRESS

**I cannot get to the index or thumbnail links via keyboard.**
I can tab through the UV tabs for Index and Thumbnails in combination with keyboard and voiceover. If I am only using the keyboard clicking enter on the tab does not switch tab view. I am able to switch tab view by control+option+spacebar

**There are places I lose my focus indicator, but it doesn't seem to be consistent; I think JS adds and removed phantom elements at times.**
Not able to recreate. Focus is working fine for me across browsers on a MAC

**One area that seems like it loses focus is actually a useful one: tabbing past the settings / gear icon, there's no visual focus indicator, but the focus is in the main viewing area. While here you can use the keyboard to pan and zoom, which is critically important if you know that's where your focus is.**
Issue still exist. Not sure what it is highlighting in between gear icon and plus icon.

**We should provide a link of keyboard shortcuts - for instance, the arrow keys pan, plus and minus zoom.**
I can currently use the plus minus and rotate icons to zoom in and out and rotate.

**Disabled buttons don't announce that they're disabled**
Using VoiceOver, I did not hear any disabled buttons announced. However, I dont know what buttons are disabled. Anyone else can confirm this?
